src/server/saved_objects/client/saved_objects_client.test.js

```diff
@@ -1,643 +1,96 @@
-import sinon from 'sinon';
-import { delay } from 'bluebird';
 import { SavedObjectsClient } from './saved_objects_client';
-import * as getSearchDslNS from './lib/search_dsl/search_dsl';
-import { getSearchDsl } from './lib';
-import elasticsearch from 'elasticsearch';
-
-describe('SavedObjectsClient', () => {
-  const sandbox = sinon.sandbox.create();
-
-  let callAdminCluster;
-  let onBeforeWrite;
-  let savedObjectsClient;
-  const mockTimestamp = '2017-08-14T15:49:14.886Z';
-  const mockTimestampFields = { updated_at: mockTimestamp };
-  const searchResults = {
-    hits: {
-      total: 3,
-      hits: [{
-        _index: '.kibana',
-        _type: 'doc',
-        _id: 'index-pattern:logstash-*',
-        _score: 1,
-        _source: {
-          type: 'index-pattern',
-          ...mockTimestampFields,
-          'index-pattern': {
-            title: 'logstash-*',
-            timeFieldName: '@timestamp',
-            notExpandable: true
-          }
-        }
-      }, {
-        _index: '.kibana',
-        _type: 'doc',
-        _id: 'config:6.0.0-alpha1',
-        _score: 1,
-        _source: {
-          type: 'config',
-          ...mockTimestampFields,
-          config: {
-            buildNum: 8467,
-            defaultIndex: 'logstash-*'
-          }
-        }
-      }, {
-        _index: '.kibana',
-        _type: 'doc',
-        _id: 'index-pattern:stocks-*',
-        _score: 1,
-        _source: {
-          type: 'index-pattern',
-          ...mockTimestampFields,
-          'index-pattern': {
-            title: 'stocks-*',
-            timeFieldName: '@timestamp',
-            notExpandable: true
-          }
-        }
-      }]
-    }
+import { SavedObjectsRepository } from './lib/repository';
+jest.mock('./lib/repository');
+
+let mockSavedObjectsRepository;
+beforeEach(() => {
+  mockSavedObjectsRepository = {
```

Comment:
Can you do this all in a `setupMockRepository` function instead that you call from each test? It might not be super important for this particular test, but in general it makes it easier to track down exactly what is happening when you're exploring a test file from an individual test rather than from top to bottom, which is common when you're debugging a given test failure.

Reply:
Sure, I can do that.